src/apps/chahub/tests/test_chahub_utils.py

```diff
@@ -1,46 +1,38 @@
-# from competitions.models import Submission as CompetitionSubmission, Competition, Phase as CompetitionPhase, CompetitionParticipant
-# # from apps.chahub.tasks import send_chahub_general_stats
-#
-#
-# class ChahubUtillityTests(TestCase):
-#     def setUp(self):
-#         self.user = ClUser.objects.create_user(username="user", password="pass")
-#         self.competition = Competition.objects.create(
-#             title="Test Competition",
-#             created_by=self.user,
-#             # modified_by=self.user,
-#             published=True,
-#         )
-#         self.participant = CompetitionParticipant.objects.create(
-#             user=self.user,
-#             competition=self.competition,
-#             status='approved'
-#             # status=ParticipantStatus.objects.get_or_create(name='approved', codename=ParticipantStatus.APPROVED)[0],
-#         )
-#         self.phase = CompetitionPhase.objects.create(
-#             competition=self.competition,
-#             index=1,
-#             start=datetime.datetime.now() - datetime.timedelta(days=30),
-#         )
-#
-#     @override_settings(CHAHUB_API_URL='http://host.docker.internal/')
-#     def test_send_to_chahub_utillity(self):
-#         # with mock.patch('apps.web.models.CompetitionSubmission.send_to_chahub') as send_to_chahub_mock:
-#         with mock.patch('chahub.utils.send_to_chahub') as send_to_chahub_mock:
-#             send_to_chahub_mock.return_value = None
-#             # Calling this as a function instead of a task?
-#             # send_chahub_general_stats()
-#             # attempts to send to Chahub once
-#             # We succesfully sent data to Chahub
-#             assert send_to_chahub_mock.called
-#
+from unittest import mock
+
+from django.conf import settings
+from django.http.response import HttpResponseBase
+from django.test import TestCase
+
+from factories import UserFactory
+
+
+class ChaHubUtilityTests(TestCase):
+    def setUp(self):
+        settings.PYTEST_FORCE_CHAHUB = True
+        # set the url to localhost for tests
+        settings.CHAHUB_API_URL = 'localhost/'
+        self.user = UserFactory.build(username='norm')
+
+    def tearDown(self):
+        settings.PYTEST_FORCE_CHAHUB = False
+        settings.CHAHUB_API_URL = None
+
+    def mock_chahub_save(self, obj):
+        with mock.patch('apps.chahub.utils.requests.post') as requests_post_mock:
+            requests_post_mock.return_value = HttpResponseBase(status=201)
+            requests_post_mock.return_value.content = ''
+            with mock.patch('apps.chahub.utils.requests.put') as requests_put_mock:
+                requests_put_mock.return_value = HttpResponseBase(status=201)
+                requests_put_mock.return_value.content = ''
+                obj.save()
+            return requests_post_mock, requests_put_mock
+
+    def test_send_to_chahub_posts_new_data_and_puts_updated_data(self):
+        post_resp, put_response = self.mock_chahub_save(self.user)
```

Comment:
why do we need to check post vs put?

Reply:
basically just making sure the update vs create was being handled 'correctly'? Tyler said something about Chahub needing put instead of post, so I'm testing that it does what it is supposed to when it is supposed to. (with my limited understanding of chahub)

Reply:
I see, I believe just post _should_ work and is what we expect to be doing (since it does `update_or_create`). @Tthomas63 want to weigh in on this one?

Reply:
I forgot exactly where we landed on this, but I believe on Chahub's side we should handle updating/creating through `POST`. This isn't necessarily correct, but because we do bulk operations, we need to handle existing/new competitions in the same batch. (Create or update)